packages/framework/src/Command/CheckRedisCommand.php

```diff
@@ -2,30 +2,55 @@
 
 namespace Shopsys\FrameworkBundle\Command;
 
+use Redis;
+use Shopsys\FrameworkBundle\Component\Redis\RedisFacade;
 use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Webmozart\Assert\Assert;
 
 class CheckRedisCommand extends Command
 {
+    protected const RETURN_CODE_OK = 0;
+    protected const RETURN_CODE_ERROR = 1;
+
     /**
      * @var string
      */
     protected static $defaultName = 'shopsys:redis:check-availability';
 
     /**
+     * @deprecated This property is deprecated since SSFW 7.3
      * @var \Redis[]
      */
     protected $cacheClients;
 
     /**
-     * @param \Redis[] $cacheClients
+     * @var \Shopsys\FrameworkBundle\Component\Redis\RedisFacade|null
      */
-    public function __construct(array $cacheClients)
-    {
-        $this->cacheClients = $cacheClients;
+    protected $redisFacade;
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Redis\RedisFacade|\Redis[] $redisFacadeOrClients
+     */
+    public function __construct($redisFacadeOrClients)
+    {
         parent::__construct();
+
+        if (!$redisFacadeOrClients instanceof RedisFacade) {
```

Comment:
It is weird to have negation in if when there is also else present

Reply:
You're right, especially if the else part actually contains the expected (non-deprecated) execution path. Fixed in 32626a9.